hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -291,59 +284,51 @@ public void checkState() {
     }
   }
 
-  //////////////////////////////////////////////////////////////////////////////
-
-  //
-  // NOTE: This method duplicates those ones of the HoodieRecordPayload and are placed here
-  //       for the duration of RFC-46 implementation, until migration off `HoodieRecordPayload`
-  //       is complete
-  //
-  public abstract HoodieRecord mergeWith(HoodieRecord other, Schema readerSchema, Schema writerSchema) throws IOException;
+  /**
+   * Get column in record to support RDDCustomColumnsSortPartitioner
+   */
+  public abstract Object getRecordColumnValues(Schema recordSchema, String[] columns, boolean consistentLogicalTimestampEnabled);
 
-  public abstract HoodieRecord rewriteRecord(Schema recordSchema, Schema targetSchema, TypedProperties props) throws IOException;
+  /**
+   * Support bootstrap.
+   */
+  public abstract HoodieRecord mergeWith(HoodieRecord other, Schema targetSchema) throws IOException;
```

Comment:
What this method does is to stitch two record. The logic of bootstrap merge is fixed. We should not let users customize the implementation of this method, right?

Reply:
I see what you're saying. This method is confusing though, since it's also called merge. 

Ideally, we shouldn't even have it in the interface (since it's not generic enough). Can we try to extract it out?

Reply:
However, this method is strong related to data types. RowData, Arrow also need different implement. So I put it into HoodieRecord interface.

Reply:
Understood. Let's keep it for now, but just rename it to `joinWith` to avoid confusion